### ISSUE TITLE
Warn pipefail does not break in case of a background process

### ISSUE THREAD

**Issue by Li0liQ** [koalaman/shellcheck#2045]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
set -o pipefail
set -o errexit
false | tee &
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

`pipefail` will not produce an error in case of a background process. Consider returning an exit code explicitly
   ```sh
#!/usr/bin/env bash
set -o pipefail
set -o errexit
{ 
    false | tee
    exit "${PIPESTATUS[0]}"
} &
```
